ui/app/src/components/DuplicateSiteDialog/DuplicateSiteDialogContainer.tsx

```diff
@@ -0,0 +1,261 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { DuplicateSiteState, LookupTable } from '../../models';
+import useSpreadState from '../../hooks/useSpreadState';
+import { DialogBody } from '../DialogBody';
+import Box from '@mui/material/Box';
+import DuplicateForm from './DuplicateForm';
+import { DialogFooter } from '../DialogFooter';
+import Button from '@mui/material/Button';
+import { FormattedMessage } from 'react-intl';
+import PrimaryButton from '../PrimaryButton';
+import React, { MouseEvent, useEffect, useMemo, useRef, useState } from 'react';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import IconButton from '@mui/material/IconButton';
+import EditIcon from '@mui/icons-material/Edit';
+import { keyframes } from 'tss-react';
+import { fadeIn } from 'react-animations';
+import { ApiResponseErrorState } from '../ApiResponseErrorState';
+import { CreateSiteDialogLoader } from '../CreateSiteDialog';
+import { duplicate, fetchLegacySite } from '../../services/sites';
+import { setSiteCookie } from '../../utils/auth';
+import { getSystemLink } from '../../utils/system';
+import useEnv from '../../hooks/useEnv';
+import { nnou } from '../../utils/object';
+import { Subscription } from 'rxjs';
+import useMount from '../../hooks/useMount';
+
+interface DuplicateSiteDialogContainerProps {
+  site: DuplicateSiteState;
+  setSite: (site: Partial<DuplicateSiteState>) => void;
+  handleClose(event?: React.MouseEvent, reason?: string): void;
+}
+
+export function DuplicateSiteDialogContainer(props: DuplicateSiteDialogContainerProps) {
+  const { site, setSite, handleClose } = props;
+  const [apiState, setApiState] = useSpreadState({
+    duplicatingSite: false,
+    error: null
+  });
+  const { authoringBase, useBaseDomain } = useEnv();
+  const fieldsErrorsLookup: LookupTable<boolean> = useMemo(() => {
+    return {
+      sourceSiteId: !site.sourceSiteId,
+      siteName: !site.siteName || site.siteNameExist,
+      siteId: !site.siteId || site.siteIdExist || site.invalidSiteId,
+      description: false,
+      gitBranch: false
+    };
+  }, [site]);
+  const [sourceSiteHasBlobStores, setSourceSiteHasBlobStores] = useState(null);
+  const finishRef = useRef(null);
+  const siteDuplicateSubscription = useRef<Subscription>();
+
+  const validateForm = () => {
+    return !(
+      !site.sourceSiteId ||
+      !site.siteId ||
+      site.siteIdExist ||
+      !site.siteName ||
+      site.siteNameExist ||
+      site.invalidSiteId
+    );
+  };
+
+  const duplicateSite = () => {
+    setApiState({ duplicatingSite: true });
```

Comment:
It is blocked while submitting because `hasPendingChanges` is true, though I'll set the `isSubmitting` for it to be more clear. Still, dialog can be closed (if clicking close in confirmation) and it'll still create in background and display the `site ready` message

Reply:
It shouldn't be closable (backdrop, escape, close) unless by the explicit create in background action.